src/lib/seam/connect/seam-http-request.ts

```diff
@@ -0,0 +1,115 @@
+import type { Method } from 'axios'
+
+import type { Client } from './client.js'
+import { SeamHttpActionAttempts } from './index.js'
+import type { SeamHttpRequestOptions } from './options.js'
+import { resolveActionAttempt } from './resolve-action-attempt.js'
+
+export interface SeamHttpRequestParent {
+  readonly client: Client
+  readonly defaults: Required<SeamHttpRequestOptions>
+}
+
+export interface SeamHttpRequestConfig<TBody> {
+  url?: string
+  method?: Method
+  params?: any
+  data?: TBody
+}
+
+export type ResponseFromSeamHttpRequest<T> =
+  T extends SeamHttpRequest<any, infer TResponse, infer TResourceKey>
+    ? TResourceKey extends keyof TResponse
+      ? TResponse[TResourceKey]
+      : undefined
+    : never
+
+export class SeamHttpRequest<
+  const TBody,
+  const TResponse,
+  const TResourceKey extends keyof TResponse | undefined,
+> implements
+    PromiseLike<
+      TResourceKey extends keyof TResponse ? TResponse[TResourceKey] : undefined
+    >
+{
+  readonly parent: SeamHttpRequestParent
+  readonly config: SeamHttpRequestConfig<TBody>
+  readonly resourceKey: TResourceKey
+  readonly options: Pick<SeamHttpRequestOptions, 'waitForActionAttempt'>
+
+  constructor(
+    parent: SeamHttpRequestParent,
+    config: SeamHttpRequestConfig<TBody>,
+    resourceKey: TResourceKey,
+    options: Pick<SeamHttpRequestOptions, 'waitForActionAttempt'> = {},
+  ) {
+    this.parent = parent
+    this.config = config
+    this.resourceKey = resourceKey
+    this.options = options
+  }
```

Comment:
Can we make all of these private?
```suggestion
  #parent: SeamHttpRequestParent
  #config: SeamHttpRequestConfig<TBody>
  #resourceKey: TResourceKey
  #options: Pick<SeamHttpRequestOptions, 'waitForActionAttempt'>

  constructor(
    parent: SeamHttpRequestParent,
    config: SeamHttpRequestConfig<TBody>,
    resourceKey: TResourceKey,
    options: Pick<SeamHttpRequestOptions, 'waitForActionAttempt'> = {},
  ) {
    this.#parent = parent
    this.#config = config
    this.#resourceKey = resourceKey
    this.#options = options
  }
```

Reply:
yep, I wasn't sure how our compilation step would treat this / what our browser support looks like.

Reply:
I think it should be ok. If we have concerns, we can use the `private` keyword instead.